bigtop-packages/src/common/apex/do-component-build

```diff
@@ -28,6 +28,9 @@ command -v mvn > /dev/null 2>&1 || {
 # Build apex
 mvn clean package -DskipTests "$@"
 
+#Find hadoop.version from pom.xml
+APEX_HADOOP_VERSION=$(grep -oPm1 "(?<=<hadoop.version>)[^<]+" pom.xml)
```

Comment:
I think what @tweise and @c0s are saying is that you need to pass whatever flags are required to apex build so that it DOES pick up Hadoop version that is coming from Bigtop BOM. That's step #0 

Then you can decide whether you want to exclude the JAR or not -- but it HAD to build with Hadoop that is enforced by Bigtop stack.

Does it make it clear?